ZIP 0: ZIP Process

acityinohio opened the pull request:
Submitting a PR for an updated ZIP process guideline.

As part of this proposal, I'd also support the zips repo being transferred to the [Zcash Foundation GitHub org](https://github.com/ZcashFoundation), with all the links being appropriately updated.

leto commented:
:-1: to breaking all existing links to current ZIPs

Honestly, it's hard to see the motivation for this change in process, when actually live ZIPs sit languishing, waiting for review: https://github.com/zcash/zips/pull/199

daira commented:
@leto A repo move will cause all links to be redirected.

acityinohio commented:
I'd be in favor of keeping it in the the [Zcash org](https://github.com/zcash) if a) org membership is public and b) the Foundation has an admin within the org. Barring that I'd prefer the ZIPs to live in a repo outside Zcash Company control, even if it means losing elegant namespacing.

str4d commented:
Renaming and changing the file in the same commit makes it very hard to review the changes from the previous version, particularly as the new file has no changelog, and there aren't any meaningful commit messages. Is this supposed to be a from-scratch rewrite? If so, it should probably have been proposed as its own draft, rather than deleting the previous one.

acityinohio commented:
@str4d thanks for the feedback, I'll update it while retaining the old draft (and I think I'm going to switch it back to `.rst` since all the other ZIPs are included as `.rst` files and markup is not a bikeshed I'm willing to die on :smile: )

defuse commented:
Looks good! Although I would want to see a  "Security & Privacy Considerations" item added to the list of parts a ZIP should have. This section would document known implementation pitfalls, or at least some text demonstrating that the champions have considered the security and privacy consequences of their proposal. [RFC 3552 - Guidelines for Writing RFC Text on Security Considerations](https://tools.ietf.org/html/rfc3552) might be useful; I haven't read it so I don't know if it's any good.

acityinohio commented:
ACK @daira, thanks! Also added a short `Security and Privacy Considerations` section per @defuse's advice.

mineZcash commented:
I would suggest that some sort of "plain English" explanation of what the ZIP process is and how the average person could contribute to the process be posted somewhere.  My gut tells me that many of the people are not commenting here or on the forum because they don't understand the process. 

Ditto for the thread that mms posted;  https://forum.zcashcommunity.com/t/call-for-nu3-zips-and-network-upgrade-pipeline-process-changes/32749

kobigurk commented:
Small comment - the ZIP Status Field section doesn't contain an Implemented status, and there are references to this status in the workflow.

rex4539 commented:
Nit: I'm not keen on using the word "Champion". It's kind of cringeworthy :)

acityinohio commented:
Good idea @mineZcash, @sonyamann is working on a blog post/forum post to help get more comments and describe the process in simpler terms.

@kobigurk nice catch! Added an Implementation description in latest commit, thanks.

@rex4539 I hear you, I wanted to confer something that had more weight/ownership than "author"...would "Owner" be a better label? Very open to suggestions here.

rex4539 commented:
I'm definitely leaning more towards "Owner" than "Champion".

str4d commented on zip-0000.rst:
```diff
@@ -0,0 +1,578 @@
+among the Editors. If it's a Standards Track ZIP, upon changing status to
+Proposed the Editors will add the optional ``Network Upgrade`` header
+to the preamble, indicating the intent for the ZIP to be implemented in
+the specified network upgrade. (All \*Network Upgrade\* schedules will be
```
```suggestion
the specified network upgrade. (All ``Network Upgrade`` schedules will be
```

str4d commented on zip-0000.rst:
```diff
@@ -0,0 +1,578 @@
+  ZIP.
+
+* Proposed: Typically the stage after Draft, added to a ZIP after
+  consideration and feedback from the community. The ZIP Editors must
```
This text appears to indicate that the ZIP may not have reached rough consensus before reaching this stage, but in the Specification section below it is clear that rough consensus is required. Perhaps:

```suggestion
  consideration, feedback, and rough consensus from the community. The ZIP Editors must
```

str4d commented on zip-0000.rst:
```diff
@@ -0,0 +1,578 @@
+
+ZIPs may include auxiliary files such as diagrams. Auxiliary files
+should be included in a subdirectory for that ZIP; that is, any ZIP that
+requires more than one file should be in a subdirectory named zip-XXXX.
```
This is not a blocker, just an observation that the wording could maybe be improved. Editors will obviously ensure that the appropriate file arrangement is used for each ZIP.

acityinohio replied:
Maybe something along the lines that unless the file type requires auxiliary files to be in the same directory (a la LaTeX) the ZIP should conform to that standard? I do see what you mean about the ambiguity though.

acityinohio replied:
(tried to update it to clarify this)

daira replied:
I reworded this again to simplify it when rebasing.

acityinohio commented:
Changed "Champion" back to "Owner" per @rex4539's suggestion and accepted all of @str4d's changes (and added some clarity re: auxiliary files too @str4d).

How does everyone feel about merging, and if we're good to go, can I be added as a contributor to the ZIPs repo? (rather than moving the repo to the Foundation, per the namespace/confusion issue) Thanks for all the feedback everyone! :pray:

rex4539 commented on zip-0000.rst:
```diff
@@ -0,0 +1,583 @@
+  Updates:
+
+The Owners header lists the names and email addresses of all the
+champions/owners of the ZIP. The format of the Owners header value
```
Is "champions" purposely left here?

acityinohio replied:
Thanks for the review @rex4539 — in my search/replace I neglected to consider case-insensitivity, just pushed a commit to fix :raised_hands:

rex4539 commented on zip-0000.rst:
```diff
@@ -0,0 +1,583 @@
+
+  Random J. User <address@dom.ain>
+
+If there are multiple champions, each should be on a separate line.
```
Is "champions" purposely left here?

daira commented:
I updated the PR to account for the draft version of the ZIP having been deleted, by request of Luke Dashjr. However in doing so made a rookie error (not pulling before force-pushing). I've corrected this now but the commit attribution might be messed up slightly; sorry about that.

daira commented:
Fixed the attributions.

daira commented:
@acityinohio I can't add you as a contributor because I'm not an admin for this repo, but I have no objection to you being added.

ebfull commented on zip-0000.rst:
```diff
@@ -0,0 +1,580 @@
+
+* Assign a ZIP number in the pull request.
+* Merge the pull request when it is ready.
+* List the ZIP in `README.rst <README.rst>`__
```
Currently, this repository only has a `README.md`.

daira replied:
I'm inclined to just remove this step. @acityinohio, does that sound reasonable?

acityinohio replied:
Totally works for me, updated with that step removed.
